lib/index.js

```diff
@@ -130,4 +133,87 @@ Dynamoose.prototype.revertDDB = function () {
   this.dynamoDB = null;
 };
 
+Dynamoose.prototype.transaction = async function(items, options, next) {
+  debugTransaction('Run Transaction');
+  const deferred = Q.defer();
+
+  options = options || {};
+  if(typeof options === 'function') {
+    next = options;
+    options = {};
+  }
+
+  if(items === null || items === undefined || items.length === 0) {
+    deferred.reject(new errors.TransactionError('Items required to run transaction'));
+    return deferred.promise.nodeify(next);
+  }
+
+  items = await Promise.all(items);
+  let transactionReq = {
+    TransactItems: items.map(item => {
+      const returnItem = {...item};
+      delete returnItem.$__;
+      return returnItem;
+    })
+  };
+  let transactionMethodName = items.map(obj => Object.keys(obj)[0]).every(key => key === "Get") ? "transactGetItems" : "transactWriteItems";
+  if (options.type) {
+    debugTransaction("Using custom transaction method");
+    if (options.type === "get") {
+      transactionMethodName = "transactGetItems";
+    } else if (options.type === "write") {
+      transactionMethodName = "transactWriteItems";
+    } else {
+      deferred.reject(new errors.TransactionError('Invalid type option, please pass in "get" or "write"'));
+      return deferred.promise.nodeify(next);
+    }
+  } else {
+    debugTransaction("Using predetermined transaction method");
+  }
+  debugTransaction(`Using transaction method: ${transactionMethodName}`);
+
+  const transact = () => {
+    console.log(JSON.stringify(transactionReq));
+    debugTransaction('transact', transactionReq);
+    this.dynamoDB[transactionMethodName](transactionReq, function(err, data) {
+      if(err) {
+        debugTransaction(`Error returned by ${transactionMethodName}`, err);
+        return deferred.reject(err);
+      }
+
+      debugTransaction(`${transactionMethodName} response`, data);
+
+      if(!data.Responses) {
+        return deferred.resolve();
+      }
+
+      return deferred.resolve(data.Responses.map(function (item, index) {
+        let model;
+        const TheModel = items[index].$__.newModel$;
+
+        Object.keys(item).forEach(function (prop) {
+          if (item[prop] instanceof DynamoDBSet) {
+            item[prop] = item[prop].values;
+          }
+        });
+
+        model = new TheModel(item);
+        model.$__.isNew = false;
+        debugTransaction(`${transactionMethodName} parsed model`, model);
+        return model;
+      }).filter(item => !(schema.expires && schema.expires.returnExpiredItems === false && item[schema.expires.attribute] && item[schema.expires.attribute] < new Date())));
```

Comment:
This is minor, but for _consistency_ i'd instantiate the `new Date()` once, and then filter based on it here. This will mean that any sort of nanosecond/millisecond roll-overs will have no weird effect, and returned records will just have a single date reference.

Reply:
@jutaz This code is written throughout the entire codebase. So I think a separate PR would need to address this. Not quite sure I understand what you are talking about tho. If you want to create a PR for that to change it throughout the codebase that'd be awesome! Or create an issue for this with a bit more detail on the proposed new solution.

Reply:
See #477